SourceX/discord/discord.cpp

```diff
@@ -0,0 +1,141 @@
+#include "discord.h"
+
+#include <Discord/cpp/discord.h>
+#include <chrono>
+#include <string>
+#include <sstream>
+
+#include "../types.h"
+
+namespace dvl {
+
+DiscordManager::DiscordManager()
+{
+	discord::Core::Create(DISCORD_DEVIX_APPID, DiscordCreateFlags_NoRequireDiscord, &core);
+}
+
+const char *difficulties[] = { "Normal", "Nightmare", "Hell" };
+const char *DifficultyToStr(int difficulty)
+{
+	if (difficulty < 0 || difficulty > NUM_DIFFICULTIES)
+		return "Unknown";
+	return difficulties[difficulty];
+}
+
+const char *playerClasses[] = { "Warrior", "Rogue", "Sorcerer", "Monk", "Bard", "Barbarian" };

```

Comment:
no need to redefine them and the function below is useless, just do 
```cpp
ClassStrTbl[plr[pnum]._pClass]
```
if taking name from player id or
```cpp
ClassStrTbl[classid]
```
if using class id directly

Reply:
Just noticed that Bard/Barbarian are called Rogue/Warrier in this array so I'll keep the one I had.